src/trace/implementations/merge_batcher_col.rs

```diff
@@ -62,7 +63,7 @@ where
     // which we call `lower`, by assumption that after sealing a batcher we receive no more
     // updates with times not greater or equal to `upper`.
     #[inline]
-    fn seal<B: Builder<Item=Self::Item, Time=Self::Time>>(&mut self, upper: Antichain<T>) -> B::Output {
+    fn seal<B: Builder<Item=Self::Item, Time=Self::Time>, C: FnOnce(usize, usize, usize) -> B>(&mut self, upper: AntichainRef<T>, constructor: C) -> B::Output {
```

Comment:
I'm not particularly happy about this `C` parameter, but it seemed easy to implement. Open to suggestions.

Reply:
Oh interesting. This may be another pattern to explore in the "how do we hook up various interfaces" saga. :D